salvo/src/lib/run_benchmark.py

```diff
@@ -107,6 +111,53 @@ def _generate_job_control_for_envoy_images(self) \
 
     return job_control_list
 
+  def _generate_job_control_for_binaries(self) \
+    -> List[proto_control.JobControl]:
+    """Determine the required source configurations needed for a binary benchmark.
+
+    Find the specified version of Nighthawk, Envoy, and all additional Envoy
+    commits to be tested. These will be spawned into multiple Binary Benchmark jobs.
+
+    Raises:
+      Exception: when multiple competing repositories of Envoy/Nighthawk are specified.
```

Comment:
(partly pre-existing) Can we raise our custom exception here (`BenchmarkRunnerError`) instead of the top level `Exception`?

Raising the top level `Exception` causes the users to catch at that level which would also catch and cover any other Python exception, e.g. SyntaxError or MemoryError that might not be desired.